main.py

```
@@ -0,0 +1,140 @@
+                               cookie_jar_path=COOKIE_JAR_PATH,
+                               url=GOB_URL, project='linux/kernel/git/torvalds/linux', branch='master')
+        self.message_dao = MessageDao()
+        self.archive_index = ArchiveMessageIndex(self.message_dao)
+        self.last_hash = self.message_dao.get_last_hash()
+    
+    @staticmethod
+    def remove_files(file_dir : str):
+        files = glob.glob(f'{file_dir}/*')
+        for f in files:
+            os.remove(f)
+    
+    @staticmethod
+    def split_parent_and_reply_messages(messages : List[Message]) -> Tuple[List[Message], List[Message]]:
+        ''' Splits a list of messages into parent (first email in a thread) and replies.
+        Returns a list of parents, then replies '''
+        parents : List[Message] = []
+        replies : List[Message] = []
+        for message in messages:
+                if not message.in_reply_to:
+                    parents.append(message)
+                    continue
+                replies.append(message)
+        return (parents, replies)
+    
+    def update_message_dir(self) -> List[Message]:
+        self.last_hash = fill_message_directory(GIT_PATH, FILE_DIR, self.last_hash)
+        messages = self.archive_index.update(FILE_DIR) 
+        return messages
+    
+    def upload_messages(self, messages_to_upload : List[str]):
+        for message_id in messages_to_upload:
+            try:
+                email_thread = self.archive_index.find(message_id)
+                patchset = parse_comments(email_thread)
+                self.gerrit_git.apply_patchset_and_cleanup(patchset, self.message_dao)
+                find_and_label_all_revision_ids(self.gerrit, patchset)
+                upload_all_comments(self.gerrit, patchset)
+            except ValueError:
+                print(f'Failed to find {message_id} in index.')
+                pass
+            except:
+                print('Failed because of an unexpected error.')
```

Should we show this unexpected error?
Or should we reraise the error or just not catch the error in the first place?